src/components/GenericAccount.tsx

```diff
@@ -76,42 +77,57 @@ const GenericAccount = ({ value }: Props) => {
   });
   const companionState = useBalance(targetApi, companionAddress, true);
 
+  const looseHelperAccount = () => {
+    setSelected('');
+    dispatchTransaction(TransactionActionCreators.setReceiverAddress(null));
+  };
+
   const setNativeAsTarget = () => {
+    if (selected) {
```

Comment:
could you please elaborate the purpose of setting the receiver to null at this time? have you verified if this is not breaking all the flow that comes on the other components?

Reply:
This is to make sure that when the companion options "menu" is open, the transaction cannot be submitted.
Imagine you you want to switch from companion to native or vice versa manually. If we don't set the receiver address to null, you'll be able to submit transaction even though you haven't selected the option you wanted to switch to.

I did not test flows. Just set the receiver to null (default value) in the same way the origin/companion are set on select

Reply:
but at this time the transfer button should be disabled right? or at least this shouldn't solve the issue?

Reply:
All this flow will be refactored anyway ... the only thing is to make sure that any part of the flow is broken

Reply:
As the UX was enhanced . This change is valid.